superset/security/manager.py

```diff
@@ -269,9 +269,12 @@ def datasource_access(self, datasource: "BaseDatasource") -> bool:
         :returns: Whether the use can access the Superset datasource
         """
 
-        return self.schema_access(datasource) or self.can_access(
-            "datasource_access", datasource.perm
-        )
+        try:
+            self.raise_for_access(datasource=datasource)
+        except SupersetSecurityException:
```

Comment:
If there's a different exception thrown, then this fails open. Is that secure?

Reply:
@etr2460 in theory (by construction) the `raise_for_access` method should only throw a `SupersetSecurityException` exception. Generally catching scoped (as opposed to general) exceptions is preferred.

Reply:
In theory, yes. I guess, what if there's an error from the db level or somewhere else and it throws a different exception? Should we do something instead like:
```python
try:
  self.raise_for_access(datasource=datasource)
  return True
except SupersetSecurityException:
  pass

return False
```